pkg/streamingpromql/operators/aggregations/limitklimitratio/utils.go

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package limitklimitratio
+
+import (
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+	"github.com/grafana/mimir/pkg/util/limiter"
+)
+
+type quietCloser interface {
```

Comment:
[nit] This interface doesn't seem to be used directly (ie. there are no methods that take or return this type, and no variables of this type) - is it necessary?

If not, I would move the `close()` method into the `stepCounter` and `stepArgument` interfaces and then remove this interface.